### ISSUE TITLE
delete() function

### ISSUE THREAD

**Issue by aplavin** [JuliaObjects/Accessors.jl#30]
I enjoy using `Setfield.jl` and recently heard about `Accessors.jl` as its successor.
Would like to propose a `delete` function that looks generally very useful:
```
delete( (a=1, b=2, c=3), @lens(_.a) ) == (b=2, c=3)
delete( (1,2,3), @lens(last(_)) ) == (1, 2)
delete( "path/to/file, @lens(basename(_)) ) == "path/to/"
...
```
Relevant prior work: `NamedTupleTools.delete((a=1, b=2, c=3), :a) == (b=2, c=3)`.

**Comment by jw3126** [comment]
Yes I am also occasionally missing a delete function. If you want to do a PR I am happy to help.

**Comment by aplavin** [comment]
And what about the "inverse"?
```
insert( (b=2, c=3), @lens(_.a), 1 ) == (b=2, c=3, a=1)
insert( (1,2), @lens(last(_)), 3 ) == (1, 2, 3)
...
```
I think the name `insert` is less intuitive compared to `delete`. But the functionality is just as useful.

**Comment by jw3126** [comment]
Yes currently we only support modifying existing fields of objects. Deleting and adding would be good to support. There are some design questions, but if someone starts a PR I am happy to support it.

**Comment by aplavin** [comment]
Great! Insert/delete are the two main things I’m missing in Setfield.
I’ll probably be able to start a PR, but later when I leave the hospital.

**Comment by jw3126** [comment]
> I’ll probably be able to start a PR, but later when I leave the hospital.

I hope you are well. I am currently moving and don't have a proper internet connection yet, so I might be slightly unresponsive.

**Comment by aplavin** [comment]
I came up with an implementation for both `delete` and `insert` functions. That was pretty straightforward following existing code in `Accessors.jl`. And nested lens work basically "for free".
For now, the code and examples are in a pluto notebook - please take a look: https://share.plav.in/accessors_jl.html. Let me know if you agree with the general approach.
Also looking for a better name instead of `insert`... It's not too obvious unlike `delete`.

**Comment by jw3126** [comment]
Nice to see you again @aplavin! Your code looks quite slick. Would be awesome if you turn it into a PR. I have more detailed comments, but I think it's easier to discuss these in an open PR, where it is easy to comment on specific lines of code etc.
